Desktop/html/js/analysis.js

```diff
@@ -596,9 +596,10 @@ JASPWidgets.AnalysisView = JASPWidgets.View.extend({
 		var results			= this.model.get("results");
 		var titleAnalysis	= this.model.get("title");
 
-		if (results == "" || results == null) {
+		if (results === "" || results === null) {
 			progress = this.model.get("progress");
-			if (progress.value > -1)
+
+			if (progress !== null && progress.value > -1)
 				this.updateProgressbarInResults();
 			return this;
 		}
```

Comment:
If you want you can delete these lines, as they won't be used anymore. They belong to the old style analysis only sending progress updates without the results being included. I don't think this is possible anymore with jaspResults

Reply:
Yeah but https://github.com/jasp-stats/INTERNAL-jasp/issues/1159 could make use of that really well.
It should also avoid some of the slowness thats been observed in the webengine during progressbar phase (because it keeps rerendering the results despite them not changing).